Core/DetectorSD.cpp

```diff
@@ -120,13 +120,13 @@ G4bool DetectorSD::ProcessHits(G4Step* aStep, G4TouchableHistory*)
     }
     else
     {
-    // This is multiple scattering and compton that does not come from primary gamma generated (pair creation, electron scattering, ...)
+    // This is multiple scattering and compton that does not come from primary gamma generated (pair creation, electron scattering, multiple scatterings)
       if (fHistoManager) {
         fHistoManager->AddNodeToDecayTree(fHistoManager->GetParentIDofPhoton() * PrimaryParticleInformation::kSecondaryParticleMultiplication, 
                                           aStep->GetTrack()->GetTrackID());
         fHistoManager->SetParentIDofPhoton(fHistoManager->GetParentIDofPhoton() * PrimaryParticleInformation::kSecondaryParticleMultiplication);
       }
-      newHit->SetGenGammaMultiplicity(fHistoManager->GetParentIDofPhoton() * PrimaryParticleInformation::kSecondaryParticleMultiplication);
```

Comment:
Was the previously assigned value incorrect? I am mostly asking with regard to interpreting these values in the simulations obtained with the current code before this change.

Reply:
In some sense yes. It was wrongly assigned the multiplicity based on the primary photon what complicated searching over the decay tree (for example photon with multiplicity 1 created secondary particle, then the created additional photon will be connected to hit with multiplicity 10, and not 1). It should not influence interpretation